[Swift in WebKit] Move GroupActivities code from WebKitSwift to WebKit

rr-codes opened the pull request:
#### 7587ce4857c86152a86a0ddf8fb3b1416d65d638
<pre>
[Swift in WebKit] Move GroupActivities code from WebKitSwift to WebKit
<a href="https://rdar.apple.com/151568946">rdar://151568946</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=293203">https://bugs.webkit.org/show_bug.cgi?id=293203</a>

Reviewed by Aditya Keerthi.

Move GroupActivities-related code to WebKit instead of WKS.

* Source/WebKit/Modules/Internal/WebKitInternal.h:

- Add WKGroupSession to the internal module

* Source/WebKit/UIProcess/Cocoa/GroupActivities/GroupActivitiesSessionNotifier.mm:

- Remove soft linking

* Source/WebKit/UIProcess/Cocoa/GroupActivities/WKGroupSession.h: Renamed from Source/WebKit/WebKitSwift/GroupActivities/WKGroupSession.h.
* Source/WebKit/UIProcess/Cocoa/GroupActivities/WKGroupSession.swift: Renamed from Source/WebKit/WebKitSwift/GroupActivities/WKGroupSession.swift.

- Import the WebKit internal module instead of WKS
- Fix a spacing typo
- Add an empty deinit to all types to workaround a compiler bug on older Swift compilers that happens when using `@_objcImplementation`
- Remove the body of the deinit of `WKGroupSessionObserver` for several reasons: (1) Due to a likely compiler bug, there is no way of being able to access `incomingSessionsTask`, and (2) this is actually dead code, since the type is used by a singleton class that is never destroyed.

* Source/WebKit/UIProcess/Cocoa/WebKitSwiftSoftLink.h:
* Source/WebKit/UIProcess/Cocoa/WebKitSwiftSoftLink.mm:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebKitSwift/WebKitSwift.h:

- Remove soft linking

Canonical link: <a href="https://commits.webkit.org/295228@main">https://commits.webkit.org/295228@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d62b6a9c53d91e77cdb06a66e6803932193ce754

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/104476 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/24186 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/14603 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/109683 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/55147 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/24567 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/32730 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/79329 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/107482 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/19107 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/94294 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/59656 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/18896 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [❌ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/54516 "Failed to compile WebKit") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/88595 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/12424 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/112068 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/31637 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/23301 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/88369 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/32001 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/90527 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/88035 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/22413 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/32931 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/10706 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/26114 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/31564 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/36904 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/31356 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/34695 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/32916 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

axiixc commented on Source/WebKit/Platform/spi/Cocoa/GroupActivities_SPI.swiftinterface:
```diff
@@ -0,0 +1,29 @@
+public struct URLActivity : GroupActivities.GroupActivity {
+    public var webpageURL: Foundation.URL
+}
+//@_spi(Safari) extension GroupActivities.URLActivity {
```
Commented out?

axiixc commented on Source/WebKit/UIProcess/Cocoa/GroupActivities/WKGroupSession.swift:
```diff
@@ -57,13 +69,17 @@ fileprivate extension WKGroupSessionState {
     }
 
     var fallbackURL: URL? {
```
Total nit: but I find these look nicer as one line when they’re trivial like this.
```suggestion
    var fallbackURL: URL? { urlActivity.webpageURL }
```

axiixc commented on Source/WebKit/UIProcess/Cocoa/GroupActivities/WKGroupSession.swift:
```diff
@@ -57,13 +69,17 @@ fileprivate extension WKGroupSessionState {
     }
+
+#if compiler(<6.0)
+    @objc deinit { }
```
Was this mandated by the compiler? Unclear from the change log.

axiixc commented on Source/WebKit/UIProcess/Cocoa/GroupActivities/WKGroupSession.swift:
```diff
@@ -142,8 +162,10 @@ fileprivate extension WKGroupSessionState {
-    deinit {
-        incomingSessionsTask?.cancel()
+    @objc deinit {
+//        Task { @MainActor [incomingSessionsTask] in
```
Remove or uncomment. Although I’m fairly certain you don’t need an explicit actor annotation here:
```swift
@objc deinit {
    Task { await incomingSessionsTask?.cancel() }
}
```

axiixc replied:
Or potentially (to avoid capture `self` during its dealloc):
```swift
@objc deinit {
    Task { [incomingSessionsTask] in incomingSessionsTask?.cancel() }
}
```

rr-codes committed:
[Swift in WebKit] Move GroupActivities code from WebKitSwift to WebKit
rdar://151568946
https://bugs.webkit.org/show_bug.cgi?id=293203

Reviewed by Aditya Keerthi.

Move GroupActivities-related code to WebKit instead of WKS.

* Source/WebKit/Modules/Internal/WebKitInternal.h:

- Add WKGroupSession to the internal module

* Source/WebKit/UIProcess/Cocoa/GroupActivities/GroupActivitiesSessionNotifier.mm:

- Remove soft linking

* Source/WebKit/UIProcess/Cocoa/GroupActivities/WKGroupSession.h: Renamed from Source/WebKit/WebKitSwift/GroupActivities/WKGroupSession.h.
* Source/WebKit/UIProcess/Cocoa/GroupActivities/WKGroupSession.swift: Renamed from Source/WebKit/WebKitSwift/GroupActivities/WKGroupSession.swift.

- Import the WebKit internal module instead of WKS
- Fix a spacing typo
- Add an empty deinit to all types to workaround a compiler bug on older Swift compilers that happens when using `@_objcImplementation`
- Remove the body of the deinit of `WKGroupSessionObserver` for several reasons: (1) Due to a likely compiler bug, there is no way of being able to access `incomingSessionsTask`, and (2) this is actually dead code, since the type is used by a singleton class that is never destroyed.

* Source/WebKit/UIProcess/Cocoa/WebKitSwiftSoftLink.h:
* Source/WebKit/UIProcess/Cocoa/WebKitSwiftSoftLink.mm:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebKitSwift/WebKitSwift.h:

- Remove soft linking

Canonical link: https://commits.webkit.org/295228@main
